### PR TITLE
kast_red

### PR DESCRIPTION
**Pull request by tiffanyhsyu**
created a wavelength solution, fixed a bug in tracing

### CODE REVIEW

**Comment by rcooke-ast** [comment]
I'm happy to merge this. Any objections?

File: src/arflux.py [hunk 1]
```diff
@@ -111,6 +111,7 @@ def bspline_magfit(wave, flux, var, flux_std, nointerp=False, **kwargs):
             fi = scipy.interpolate.InterpolatedUnivariateSpline(wave[~bad], logivar[~bad], k=2)
             logivar[bad] = fi(wave[bad])
 
+#    xdb.set_trace()
```
Comment (profxj) [inline src/arflux.py]: remove this

File: src/arflux.py [hunk 2]
```diff
@@ -226,7 +227,10 @@ def find_standard_file(argflag, radec, toler=20.*u.arcmin, check=False):
     std_file_fmt = [1]  # 1=Calspec style FITS binary table
 
     # SkyCoord
-    obj_coord = SkyCoord(radec[0], radec[1], unit=(u.hourangle, u.deg))
+    try:
```
Comment (profxj) [inline src/arflux.py]: can take out this try, I think.

File: src/armlsd.py [hunk 3]
```diff
@@ -115,6 +116,8 @@ def ARMLSD(argflag, spect, fitsdict, reuseMaster=False):
             slf.SetFrame(slf._rordpix, rordpix, det)
             # Save QA for slit traces
             arqa.slit_trace_qa(slf, slf._mstrace[det-1], slf._lordpix[det-1], slf._rordpix[det-1], extord, desc="Trace of the slit edges")
+#            slf._qa.close()
```
Comment (profxj) [inline src/armlsd.py]: Remove this and the other slf._qa.close() lines that follow.
We may want, however, to add a try/except clause that
catches a failed run and closes the PDF on a failure.

File: src/artrace.py [hunk 4]
```diff
@@ -222,7 +222,8 @@ def trace_object(slf, det, sciframe, varframe, crmask, trim=2.0, triml=None, tri
     rec_obj_img = np.zeros((sciframe.shape[0],sciframe.shape[1],nobj))
     for o in range(nobj):
         obj = np.zeros(npix)
-        obj[objl[0]:objr[0]+1]=1
+#        obj[objl[0]:objr[0]+1]=1
```
Comment (profxj) [inline src/artrace.py]: Can remove the buggy line

File: src/artrace.py [hunk 5]
```diff
@@ -1170,6 +1171,10 @@ def model_tilt(slf, det, msarc, guesstilts=None, censpec=None, plotQA=False, ref
             # Get the size of the slit
             sz = int(np.floor(np.abs(slf._rordloc[det-1][arcdet[j],0]-slf._lordloc[det-1][arcdet[j],0])/2.0))-2
             #xtfit = np.arange(-sz,sz+1,1.0) # pixel along the arc line
+#            try:
```
Comment (profxj) [inline src/artrace.py]: remove this try

File: src/arutils.py [hunk 6]
```diff
@@ -11,6 +11,8 @@
 import arcyarc
 import warnings
 
+from xastropy.xutils import xdebug as xdb
```
Comment (profxj) [inline src/arutils.py]: comment this out

File: src/arutils.py [hunk 7]
```diff
@@ -109,9 +111,9 @@ def bspline_fit(x,y,order=3,knots=None,everyn=20,xmin=None,xmax=None,w=None,bksp
             msgs.error("No method specified to generate knots")
     # Generate spline
     try:
-        tck = interpolate.splrep( x[gd], y[gd], w=weights, k=order, t=knots)
+        tck = interpolate.splrep(x[gd], y[gd], w=weights, k=order, t=knots)
     except ValueError: # Knot problem
-        pdb.set_trace()
+        xdb.set_trace()
```
Comment (profxj) [inline src/arutils.py]: return to pdb

**Comment by profxj** [comment]
Have put in some inline comments.
Tiffany, please make those edits and we can then merge.

**Comment by profxj** [comment]
Merging..

**Pull request merged**